robotpy_ext/control/toggle.py

```diff
@@ -0,0 +1,107 @@
+from _functools import partial
+
+
+class Toggle:
+    """Utility class for button toggle
+
+        Usage:
+
+        foo = Toggle(joystick, 3)
+
+        if foo:
+            toggleFunction()
+
+        if foo.on:
+            onToggle()
+
+        if foo.off:
+            offToggle()
+    """
+    class _SteadyDebounce:
+        '''
+            Similar to ButtonDebouncer, but the output stays steady for
+            the given periodic_filter. E.g, if you set the period to 2
+            and press the button, the value will return true for 2 seconds.
+
+            Steady debounce will return true for the given period, allowing it to be
+            used with Toggle
+        '''
+
+        def __init__(self, joystick, button, period=0.5):
+            from wpilib.timer import Timer
+            '''
+            :param joystick:  Joystick object
+            :type  joystick:  :class:`wpilib.Joystick`
+            :param button: Number of button to retrieve
+            :type  button: int
+            :param period:    Period of time (in seconds) to wait before allowing new button
+                              presses. Defaults to 0.5 seconds.
+            :type  period:    float
+            '''
+            self.joystick = joystick
+            self.button = button
+            self.latest = 0
+            self.debounce_period = float(period)
+            self.timer = Timer
+
+        def set_debounce_period(self, period):
+            '''Set number of seconds to hold return value'''
+            self.debounce_period = float(period)
+
+        def get(self):
+            '''Returns the value of the joystick button. Once the button is pressed,
+            the return value will be True until the time expires
+            '''
+
+            now = self.timer.getFPGATimestamp()
+            if now - self.latest < self.debounce_period:
```

Comment:
Because I want it to return true while the time is less than the period. 

Let's say `debounce_period = 1`. If I press the button at `t = 0`, then at `t = .5` it will still return true.